NXP HAL should support IPv6 only configuration in networking middleware

**Is your enhancement proposal related to a problem? Please describe.**
I tried to compile a IPv6 only Wi-Fi application for `frdm-rw612` board. The IPv6 only snippet can be found in #87644

```
west build -p \
-d ~/zephyrproject/build/zperf-nxp \
~/zephyrproject/zephyr/samples/net/zperf/ \
-b frdm_rw612 \
-S wifi-ipv6
```
Compiling this I also needed #87573 and #87410

The building fails with this message
```
zephyrproject/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c: In function 'stop_cb':
zephyrproject/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c:990:5: warning: implicit declaration of function 'net_dhcpv4_stop'; did you mean 'net_dhcpv6_stop'? [-Wimplicit-function-declaration]
  990 |     net_dhcpv4_stop(if_handle->netif);
      |     ^~~~~~~~~~~~~~~
      |     net_dhcpv6_stop
zephyrproject/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c: In function 'net_configure_address':
zephyrproject/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c:1150:13: warning: implicit declaration of function 'net_dhcpv4_restart'; did you mean 'net_dhcpv6_restart'? [-Wimplicit-function-declaration]
 1150 |             net_dhcpv4_restart(if_handle->netif);
      |             ^~~~~~~~~~~~~~~~~~
      |             net_dhcpv6_restart
zephyrproject/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c: In function 'net_get_if_addr':
zephyrproject/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c:1201:60: error: 'struct net_if_ip' has no member named 'ipv4'; did you mean 'ipv6'?
 1201 |     struct net_if_ipv4 *ipv4 = if_handle->netif->config.ip.ipv4;
      |                                                            ^~~~
      |                                                            ipv6
zephyrproject/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c: In function 'net_get_if_ip_addr':
zephyrproject/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c:1399:60: error: 'struct net_if_ip' has no member named 'ipv4'; did you mean 'ipv6'?
 1399 |     struct net_if_ipv4 *ipv4 = if_handle->netif->config.ip.ipv4;
      |                                                            ^~~~
      |                                                            ipv6
zephyrproject/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c: In function 'net_get_if_ip_mask':
zephyrproject/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c:1408:60: error: 'struct net_if_ip' has no member named 'ipv4'; did you mean 'ipv6'?
 1408 |     struct net_if_ipv4 *ipv4 = if_handle->netif->config.ip.ipv4;
      |                                                            ^~~~
      |                                                            ipv6
[392/565] Building C object modules/hal_nxp/hal_nxp/CMakeFiles/..__modules__hal__nxp.dir/mcux/middleware/wifi_nxp/wlcmgr/wlan.c.obj
```


**Describe the solution you'd like**
As Zephyr supports IPv6 only configuration, it should be able to build also with this HAL.

--- a/drivers/wifi/nxp/src/nxp_wifi_net.c
+++ b/drivers/wifi/nxp/src/nxp_wifi_net.c
@@ -916,8 +916,10 @@ void net_stop_dhcp_timer(void)
 static void stop_cb(void *ctx)
 {
 	interface_t *if_handle = (interface_t *)net_get_mlan_handle();
-
+#if defined(CONFIG_NET_DHCPV4)
 	net_dhcpv4_stop(if_handle->netif);
+#endif
+
 #ifdef CONFIG_NXP_WIFI_IPV6
 	if (!is_sta_ipv6_connected()) {
 		(void)net_if_dormant_on(if_handle->netif);
@@ -1055,6 +1057,7 @@ int net_configure_address(struct net_ip_config *addr, void *intrfc_handle)
 
 	switch (addr->ipv4.addr_type) {
 	case NET_ADDR_TYPE_STATIC:
+#if defined(CONFIG_NET_IPV4)
 		if (addr->ipv4.address != 0) {
 			NET_IPV4_ADDR_U32(if_handle->ipaddr) = addr->ipv4.address;
 			NET_IPV4_ADDR_U32(if_handle->nmask) = addr->ipv4.netmask;
@@ -1066,6 +1069,7 @@ int net_configure_address(struct net_ip_config *addr, void *intrfc_handle)
 							&if_handle->ipaddr.in_addr,
 							&if_handle->nmask.in_addr);
 		}
+#endif
 		break;
 	case NET_ADDR_TYPE_DHCP:
 #if defined(CONFIG_WIFI_STA_AUTO_DHCPV4)
@@ -1122,11 +1126,16 @@ int net_configure_address(struct net_ip_config *addr, void *intrfc_handle)
 int net_get_if_addr(struct net_ip_config *addr, void *intrfc_handle)
 {
 	interface_t *if_handle = (interface_t *)intrfc_handle;
+
+#if defined(CONFIG_NET_IPV4)
 	struct net_if_ipv4 *ipv4 = if_handle->netif->config.ip.ipv4;
 
 	addr->ipv4.address = NET_IPV4_ADDR_U32(ipv4->unicast[0].ipv4.address);
 	addr->ipv4.netmask = ipv4->unicast[0].netmask.s_addr;
 	addr->ipv4.gw = ipv4->gw.s_addr;
+#else
+	ARG_UNUSED(if_handle);
+#endif
 
 #ifdef CONFIG_DNS_RESOLVER
 	struct dns_resolve_context *ctx;
@@ -1309,19 +1318,31 @@ static void net_clear_ipv6_ll_address(void *intrfc_handle)
 
 int net_get_if_ip_addr(uint32_t *ip, void *intrfc_handle)
 {
+#if defined(CONFIG_NET_IPV4)
 	interface_t *if_handle = (interface_t *)intrfc_handle;
 	struct net_if_ipv4 *ipv4 = if_handle->netif->config.ip.ipv4;
 
 	*ip = NET_IPV4_ADDR_U32(ipv4->unicast[0].ipv4.address);
+#else
+	ARG_UNUSED(intrfc_handle);
+	*ip = 0U;
+#endif
+
 	return WM_SUCCESS;
 }
 
 int net_get_if_ip_mask(uint32_t *nm, void *intrfc_handle)
 {
+#if defined(CONFIG_NET_IPV4)
 	interface_t *if_handle = (interface_t *)intrfc_handle;
 	struct net_if_ipv4 *ipv4 = if_handle->netif->config.ip.ipv4;
 
 	*nm = ipv4->unicast[0].netmask.s_addr;
+#else
+	ARG_UNUSED(intrfc_handle);
+	*nm = 0U;
+#endif
+
 	return WM_SUCCESS;
 }
 

